src/main/java/nwoolcan/model/brewery/production/batch/step/StepInfoImpl.java

```diff
@@ -35,33 +35,33 @@ public StepInfoImpl(final StepType stepType, final Date startDate) {
     }
 
     @Override
-    public final StepType getType() {
+    public StepType getType() {
         return this.stepType;
     }
 
     @Override
-    public final Optional<String> getNote() {
+    public Optional<String> getNote() {
         return Optional.ofNullable(this.note);
     }
 
     @Override
-    public final Result<Empty> setNote(final String note) {
+    public Result<Empty> setNote(final String note) {
         this.note = note;
```

Comment:
if null an exception has to be thrown. use Objects.requireNonNull() instead
```suggestion
        this.note = Objects.requireNonNull(note);
```